src/main/java/de/tum/in/www1/artemis/service/metis/ConversationMessagingService.java

```diff
@@ -182,7 +191,13 @@ private Conversation mayUpdateOrDeleteMessageElseThrow(Post existingMessagePost,
             throw new AccessForbiddenException("Post", existingMessagePost.getId());
         }
         else {
-            return conversationService.getConversationById(existingMessagePost.getConversation().getId());
+            var conversation = conversationService.getConversationById(existingMessagePost.getConversation().getId());
+            if (conversation instanceof Channel channel) {
+                if (channel.getIsArchived()) {
```

Comment:
Codacy warning

Reply:
Fixed in 0dd03f26a7f0b4414b762f40b6e9bbcca87798ec